aw-server/src/endpoints/hostcheck.rs

```diff
@@ -0,0 +1,45 @@
+/// Host header check needs to be performed to protect against DNS poisoning attacks.
+///
+/// Based on API key PR in [2].
+///
+/// [1]: https://github.com/ActivityWatch/activitywatch/security/advisories/GHSA-v9fg-6g9j-h4x4
+/// [2]: https://github.com/ActivityWatch/aw-server-rust/pull/185
+use rocket::http::Status;
+use rocket::request::{self, FromRequest, Request};
+use rocket::{Outcome, State};
+
+use crate::config::AWConfig;
+
+pub struct HostCheck();
+
+#[derive(Debug)]
+pub enum HostCheckError {
+    Invalid,
+}
+
+// TODO: Should this be an app-wide fairing instead? (apparently fairings can't cancel/reject requests?)
+// TODO: Use guard on any remaining sensitive endpoints
```

Comment:
If we use a fairing it will be applied to every endpoint, regardless if it is sensitive or not.

Reply:
That's the correct behavior anyway.